api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimResultTest.java

```diff
@@ -0,0 +1,115 @@
+import uk.gov.dhsc.htbhf.claimant.entitlement.VoucherEntitlement;
+import uk.gov.dhsc.htbhf.claimant.entity.Claim;
+import uk.gov.dhsc.htbhf.claimant.model.VerificationResult;
+import uk.gov.dhsc.htbhf.eligibility.model.CombinedIdentityAndEligibilityResponse;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.dhsc.htbhf.TestConstants.EXACTLY_SIX_MONTH_OLD;
+import static uk.gov.dhsc.htbhf.TestConstants.EXACTLY_THREE_YEAR_OLD;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimTestDataFactory.aClaimWithExpectedDeliveryDateAndChildrenDobs;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimTestDataFactory.aValidClaim;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.VoucherEntitlementTestDataFactory.aValidVoucherEntitlement;
+import static uk.gov.dhsc.htbhf.eligibility.model.testhelper.CombinedIdAndEligibilityResponseTestDataFactory.anIdMatchedEligibilityConfirmedUCResponseWithAllMatches;
+
+class ClaimResultTest {
+
+    @Test
+    void shouldCreateResultWithEntitlement() {
+        Claim claim = aValidClaim();
+        VoucherEntitlement voucherEntitlement = aValidVoucherEntitlement();
+        CombinedIdentityAndEligibilityResponse identityAndEligibilityResponse = anIdMatchedEligibilityConfirmedUCResponseWithAllMatches();
+
+        ClaimResult result = ClaimResult.withEntitlement(claim, voucherEntitlement, identityAndEligibilityResponse);
+
+        assertThat(result).isNotNull();
+        assertThat(result.getClaim()).isEqualTo(claim);
+        VerificationResult verificationResult = result.getVerificationResult();
+        assertThat(verificationResult.getAddressLine1Match()).isEqualTo(identityAndEligibilityResponse.getAddressLine1Match());
+        assertThat(verificationResult.getDeathVerificationFlag()).isEqualTo(identityAndEligibilityResponse.getDeathVerificationFlag());
+        assertThat(verificationResult.getEligibilityOutcome()).isEqualTo(identityAndEligibilityResponse.getEligibilityStatus());
+        assertThat(verificationResult.getEmailAddressMatch()).isEqualTo(identityAndEligibilityResponse.getEmailAddressMatch());
+        assertThat(verificationResult.getIdentityOutcome()).isEqualTo(identityAndEligibilityResponse.getIdentityStatus());
+        assertThat(verificationResult.getMobilePhoneMatch()).isEqualTo(identityAndEligibilityResponse.getMobilePhoneMatch());
+        assertThat(verificationResult.getPostcodeMatch()).isEqualTo(identityAndEligibilityResponse.getPostcodeMatch());
+        assertThat(verificationResult.getPregnantChildDOBMatch()).isEqualTo(identityAndEligibilityResponse.getPregnantChildDOBMatch());
+        assertThat(verificationResult.getQualifyingBenefits()).isEqualTo(identityAndEligibilityResponse.getQualifyingBenefits());
```

Comment:
You should be able to use `assertThat(verificationResult).isEqualToIgnoringGivenFields(identityAndEligibilityResponse, propperties)` listing the names of the properties which you don't want to compare - the objects do not need to be of the same type, it just does field by field comparison.

Reply:
Although saying that I've just spotted that `identityOutcome` is `identityStatus` on `IdentityAndEligibilityResponse` so that then might be a little less ideal if the fields don't exactly tie up.